### PR TITLE
add retry to clickstream dags

### PR DESCRIPTION
**Pull request by tylermichael**

### CODE REVIEW

File: airflow_home/utils/docker.py [hunk 1]
```diff
@@ -51,18 +51,36 @@ def create_docker_operator(params):
...
+        resources=resources)
+
+
+def create_linked_docker_operator(
+        dag, activity_list, initial_task_id, activity_tuple,
+        retries=None,
+        retry_delay=timedelta(seconds=300),
+        force_pull=None,
+        pool=None,
+        resources=None):
```
Comment (tedmiston) [inline airflow_home/utils/docker.py]: Use `retry_delay=None` in constructor in conjunction with logic suggestion below.  (This is a common pattern for substituting default values in python.)

File: airflow_home/utils/docker.py [hunk 2]
```diff
@@ -146,5 +164,13 @@ def create_linked_docker_operator(dag, activity_list, initial_task_id, activity_
...
     }
 
+    if 'retries' in dag.default_args and retries is None:
+        params['retries'] = dag.default_args['retries']
+        if retry_delay is None:
+            if 'retry_delay' in dag.default_args:
+                params['retry_delay'] = dag.default_args['retry_delay']
+            else:
+                params['retry_delay'] = retry_delay
+
```
Comment (tedmiston) [inline airflow_home/utils/docker.py]: I noticed that we have a second earlier variable also named params in the same function which can be confusing (https://github.com/astronomerio/docker-airflow/blob/6cfab29e18678dba5eef3fde8b9601757f1f346d/airflow_home/utils/docker.py).

While you are modifying `params` in this file, can you rename the second earlier one to `template_params` or `command_params` for clarity?

Comment (tylermichael) [inline airflow_home/utils/docker.py]: Yeah, I agree. Forgot the case where they are using the default and just want to change the delay.

Comment (tylermichael) [inline airflow_home/utils/docker.py]: This whole section: https://github.com/astronomerio/docker-airflow/blob/6cfab29e18678dba5eef3fde8b9601757f1f346d/airflow_home/utils/docker.py#L115-L119 isn't even used in the rest of the function.

I can remove it if you'd like.

Comment (tedmiston) [inline airflow_home/utils/docker.py]: It's used on this line - https://github.com/astronomerio/docker-airflow/blob/6cfab29e18678dba5eef3fde8b9601757f1f346d/airflow_home/utils/docker.py#L160

Confusing to read I know, but line 160 there sets `params['params'] = params` (lol) where `params` on the right is the one created on line 119 and params on the left is the one created on line 153.

**Comment by perryao** [comment]
what are your guys' latest thoughts on our git flow? feature -> master, feature -> develop -> master?

**Comment by tedmiston** [comment]
@perryao I like feature --> develop --> master for mature repos.  For repos where things are still changing lots, I'm okay with feature --> master.  The most important thing to me is isolating change patches with separate branches.

**Comment by tedmiston** [comment]
@tylermichael looks good to me — feel free to merge and deploy

**Pull request merged**